src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/RazorCSharpStatementKeywordFoldingProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Folding;
+
+internal sealed class RazorCSharpStatementKeywordFoldingProvider : AbstractSyntaxNodeFoldingProvider<CSharpCodeBlockSyntax>
```

Comment:
Does this not need to be registered somewhere in the LSP server?

Reply:
OMG 🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️

I even tested this is VS _after_ writing on the issue that it works in VS regardless of this code.

Thank you!